src/test/java/org/elasticsearch/search/highlight/HighlighterSearchTests.java

```diff
@@ -659,24 +659,36 @@ public void testPlainHighlighterDocumentAnalyzer() throws Exception {
                     .field("type", "string")
                     .field("index", "not_analyzed")
                 .endObject()
-                .startObject("text")
+                .startObject("field1")
                     .field("type", "string")
                 .endObject()
+                .startObject("field2")
+                    .field("type", "string")
+                    .field("analyzer", "standard")
+                .endObject()
             .endObject()
             .endObject().endObject()).execute().actionGet();
         ensureYellow();
 
         index("test", "type1", "1",
                 "language_analyzer", "english",
-                "text", "Look at me, I'm eating cars.");
+                "field1", "Look at me, I'm eating cars.",
+                "field2", "Look at me, I'm eating cars.");
         refresh();
 
         SearchResponse response = client().prepareSearch("test")
-                .setQuery(QueryBuilders.matchQuery("text", "car"))
+                .setQuery(QueryBuilders.matchQuery("field1", "car"))
+                .addHighlightedField(
+                        new HighlightBuilder.Field("field1").preTags("<1>").postTags("</1>").requireFieldMatch(true))
+                .get();
+        assertHighlight(response, 0, "field1", 0, 1, equalTo("Look at me, I'm eating <1>cars</1>."));
+
+        response = client().prepareSearch("test")
+                .setQuery(QueryBuilders.matchQuery("field2", "cars"))
                 .addHighlightedField(
-                        new HighlightBuilder.Field("text").preTags("<1>").postTags("</1>").requireFieldMatch(true))
+                        new HighlightBuilder.Field("field2").preTags("<1>").postTags("</1>").requireFieldMatch(true))
                 .get();
-        assertHighlight(response, 0, "text", 0, 1, equalTo("Look at me, I'm eating <1>cars</1>."));
+        assertHighlight(response, 0, "field2", 0, 1, equalTo("Look at me, I'm eating <1>cars</1>."));
```

Comment:
are you sure that this test fails without your fix? Seems like it still use the analyzer path, which used to work before too. I think we should write a different test for _analyzer without path defined in the mapping?